plugins/modules/puppet.py

```diff
@@ -101,6 +101,10 @@
       - Whether to print a transaction summary.
     type: bool
     default: false
+  waitforlock:
+    description:
+     - The maximum amount of time C(puppet) should wait for an already running C(puppet) agent to finish before starting.
```

Comment:
Can you maybe add a reference on which format is accepted here? Or mention something like
```suggestion
     - The maximum amount of time C(puppet) should wait for an already running C(puppet) agent to finish before starting.
     - If a number without unit is provided, it is assumed to be a number of seconds. Allowed units are V(m) for minutes and V(h) for hours.
```
(The `h` is a guess, please adjust if necessary :) )